doc(secret_manager): add tutorial to access rdb with credentials stored in secret manager (#3244)
scaleway/docs-content: tutorials/sem-terraform/index.mdx

---
meta:
  title: Creating and accessing a Database Instance using Secret Manager and Terraform

>   description: Discover how to create and access securely a Database Instance using Secret Manager and Terraform

Reviewer: ```suggestion
  description: Discover how to create and securely access a Database Instance using Secret Manager and Terraform
```

content:
  h1: Creating and accessing a Database Instance using Secret Manager and Terraform

>   paragraph: Discover how to create and access securely a Database Instance using Secret Manager and Terraform

Reviewer: ```suggestion
  paragraph: Discover how to create and securely access a Database Instance using Secret Manager and Terraform
```

tags: secret-manager terraform secret database-instance access
categories:
  - security-identity
  - secret-manager
  - managed-databases
dates:

>   validation: 2023-07-10

Reviewer: ```suggestion
  validation: 2024-05-30
```

  posted: 2024-05-30
---

In this tutorial, you will use a Terraform configuration file that builds your entire infrastructure such as a [Database Instance](/managed-databases/postgresql-and-mysql/concepts/#database-instance), a [secret](/identity-and-access-management/secret-manager/concepts/#secret) and a [version](/identity-and-access-management/secret-manager/concepts/#version) in which your database credentials are stored. You will then be able to access your database credentials securely from the Scaleway console.

<Macro id="iam-requirements" />

> <Message type="requirement">

Reviewer: ```suggestion
```

>  - You have an account and logged into the [Scaleway console](https://console.scaleway.com/)

Reviewer: ```suggestion
 - A Scaleway account logged into the [console](https://console.scaleway.com/)
```

>  - You have created an [API key](/identity-and-access-management/iam/how-to/create-api-keys/)

Reviewer: ```suggestion
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
 - A valid [API key](/identity-and-access-management/iam/how-to/create-api-keys/)
```

>  - You have [configured your environment variables](/developer-tools/devtools/reference-content/environment-variables/#recommended-scaleway-configuration-file)

Reviewer: ```suggestion
 - [Configured your environment variables](/developer-tools/devtools/reference-content/environment-variables/#recommended-scaleway-configuration-file)
```

>  - [Configured your environment variables](/developer-tools/devtools/reference-content/environment-variables/#recommended-scaleway-configuration-file)

Reviewer 2: Commentaire pour moi-même: le lien renvoie vers une 404

>  - [Configured your environment variables](/developer-tools/devtools/reference-content/environment-variables/#recommended-scaleway-configuration-file)

Reviewer 2: ```suggestion
```

</Message>

## Clone the GitHub repository

> We have created a [GitHub repository](https://github.com/scaleway/tutorial-sem-terraform) that contains all the code you need to deploy your infrastructure.

Reviewer: ```suggestion
The Scaleway [GitHub repository](https://github.com/scaleway/tutorial-sem-terraform) contains all the code you need to deploy your infrastructure.
```

Open a terminal and clone the repository to your local machine using the following command:

        ```bash
        git clone git@github.com:scaleway/tutorial-sem-terraform.git
        ```

> ### Files and directories

Reviewer: ```suggestion
### Understanding the files and directories
```
